provider/defaultingress/ingress.go

```diff
@@ -0,0 +1,269 @@
+	string(cmv1.WildcardPolicyWildcardsAllowed)}
+var DefaultWildcardPolicy = cmv1.WildcardPolicyWildcardsDisallowed
+
+var ValidNamespaceOwnershipPolicies = []string{string(cmv1.NamespaceOwnershipPolicyStrict),
+	string(cmv1.NamespaceOwnershipPolicyInterNamespaceAllowed)}
+var DefaultNamespaceOwnershipPolicy = cmv1.NamespaceOwnershipPolicyStrict
+
+func IngressResource() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Description: "Unique identifier of the ingress.",
+			Computed:    true,
+			Optional:    true,
+			PlanModifiers: []planmodifier.String{
+				// This passes the state through to the plan, preventing
+				// "known after apply" since we know it won't change.
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"route_selectors": schema.MapAttribute{
+			Description: "Route Selectors for ingress. Format should be a comma-separated list of 'key=value'. " +
+				"If no label is specified, all routes will be exposed on both routers." +
+				"For legacy ingress support these are inclusion labels, otherwise they are treated as exclusion label.",
+
+			ElementType: types.StringType,
+			Optional:    true,
+			Validators:  []validator.Map{NotEmptyMapValidator()},
+		},
+		"excluded_namespaces": schema.ListAttribute{
+			Description: "Excluded namespaces for ingress. Format should be a comma-separated list 'value1, value2...'. " +
+				"If no values are specified, all namespaces will be exposed.",
+			ElementType: types.StringType,
+			Optional:    true,
+			Validators: []validator.List{
+				listvalidator.SizeAtLeast(1),
+			},
+		},
+		"route_wildcard_policy": schema.StringAttribute{
+			Description: fmt.Sprintf("Wildcard Policy for ingress. Options are %s. Default is '%s'.",
+				strings.Join(ValidWildcardPolicies, ","), DefaultWildcardPolicy),
+			Optional:   true,
+			Computed:   true,
+			Validators: []validator.String{attrvalidators.EnumValueValidator(ValidWildcardPolicies)},
+		},
+		"route_namespace_ownership_policy": schema.StringAttribute{
+			Description: fmt.Sprintf("Namespace Ownership Policy for ingress. Options are %s. Default is '%s'.",
+				strings.Join(ValidNamespaceOwnershipPolicies, ","), DefaultNamespaceOwnershipPolicy),
+			Optional:   true,
+			Computed:   true,
+			Validators: []validator.String{attrvalidators.EnumValueValidator(ValidNamespaceOwnershipPolicies)},
+		},
+		"cluster_routes_hostname": schema.StringAttribute{
+			Description: "Components route hostname for oauth, console, download.",
+			Optional:    true,
+		},
+		"cluster_routes_tls_secret_ref": schema.StringAttribute{
+			Description: "Components route TLS secret reference for oauth, console, download.",
+			Optional:    true,
+		},
+	}
+}
+
+func PopulateDefaultIngress(ctx context.Context, state *DefaultIngress,
+	client *cmv1.IngressesClient) (*DefaultIngress, error) {
+
+	// in case default ingress is not part of state no need to populate it
```

Comment:
Bringing default ingress requires one more api call that has no point if you don't need it.
Another problem will be computed/optional that doesn't work good on custom structs and creates issues actually. So my idea was just to not bring it in case it was not required

Reply:
But what about the "import" case, what happen if you want to import the cluster with this value?

Reply:
If you will need to change anything in default ingress, you will add this value in order to update it.

Reply:
The issue is that you can't set custom types as optional/computed as you can't have them unknown in case you know how to do it i will be glad to change it

Reply:
We are accepting the values as part of the cluster creation request body, which means that for TF it needs to be in place for the state. Prior to this there was no possibility of changing the default values, so it was available similarly to the default machine pool (worker) where it would be a sub resource.